File: src/main/java/io/pageobject/generator/name/PostfixNamePostProcessor.java
package io.pageobject.generator.name;

import org.jsoup.nodes.Element;

import java.util.Arrays;

import static com.google.common.base.Strings.isNullOrEmpty;

public abstract class PostfixNamePostProcessor implements NamePostProcessor {

    private final String postFix;
    private final String[] postFixExceptions;

    public PostfixNamePostProcessor(String postFix, String[] postFixExceptions) {
        this.postFix = postFix;
        this.postFixExceptions = postFixExceptions;
    }

    @Override
    public String process(String name, Element element) {
        if (!isNullOrEmpty(name) && !nameEndsWithAnyOfTheExceptionPrefixes(name)) {
            name += postFix;
        }

        return name;
    }

    private boolean nameEndsWithAnyOfTheExceptionPrefixes(String name) {
        return Arrays.stream(postFixExceptions).anyMatch(name::endsWith);
    }

}

File: lib/name/postfix-name-post-processor.js
"use strict";

const _ = require('lodash');

class PostfixNamePostProcessor {

  constructor(postfix, postfixExceptions) {
    this._postfix = postfix;
    this._postfixExceptions = postfixExceptions;

    if (this.isMatchingElement === undefined) {
      throw new TypeError('Must override isMatchingElement');
    }
  }

  process(name, element) {
    if (!_.isEmpty(name) && !nameEndsWithAnyOfTheExceptionPrefixes.call(this, name)) {
      name += this._postfix;
    }

    return name;
  }

}

function nameEndsWithAnyOfTheExceptionPrefixes(name) {
  return _.some(this._postfixExceptions, (postfix) => {
    return name.endsWith(postfix)
  });
}

module.exports = PostfixNamePostProcessor;